api/v1/services/EventService.js

```diff
@@ -0,0 +1,153 @@
+                        event_obj.attributes.locations = result;
+                        return LocationEvents.addRecordModels(result || [], [event_obj]);
+                })
+                .then(function (result) {
+                        return event_obj;
+                });
+};
+
+function appendEvent(location) {
+        return LocationEvents.getEventsFromLocationId(location.getEvents('id'))
+                .then(function(result) {
+                        return _Promise.map(result.models, function(mapModel) {
+                                return Event.findById(mapModel.get('eventId'));
+                        });
+                });
+}
+
+function appendLocation(event) {
+        return LocationEvents.getLocationsFromEventId(event.get('id'))
+                .then(function (result) {
+                        return _Promise.map(result.models, function(mapModel) {
+                                return Location.findById(mapModel.get('locationId'));
+                        })
+                                .then(function(locations) {
+                                        event.attributes.locations = locations;
+                                        return event;
+                                });
+                });
+}
+
+/**
+ * Find event and corresponding locations.
+ */
+module.exports.findEventById = function (id) {
+        return Event.findById(id)
+                .then(function (result) {
+                        if (_.isNull(result)) {
+                                var message = "The Event with the given ID could not be found.";
+                                var source = "id";
+                                throw new errors.NotFoundError(message, id);
+                        }
+
+                        result.attributes.locations = [];
+                        return appendLocation(result);
+                });
+};
+
+module.exports.findLocationById = function(id) {
+        return Location.findById(id)
+                .then(function (result) {
+                        if (_.isNull(result)) {
+                                var message = "The Event with the given ID could not be found.";
+                                var source = "id";
+                                throw new errors.NotFoundError(message, id);
+                        }
+
+                        result.attributes.events = [];
+                        return appendEvent(result);
+                });
+};
+
+module.exports.findByUpdated = function (unix_timestamp) {
+        return Event.findByUpdated(unix_timestamp)
+                .then(function (events) {
+                        return _Promise.map(events.models, function(event) {
+                                event.attributes.locations = [];
+                                return appendLocation(event);
+                        });
+                });
+};
+
+module.exports.createLocation = function (name, short_name, latitude, longitude) {
```

Comment:
For creating a new location model shouldn't it forge a location, validate it, catch the checkiterror, then save the model?